### PR TITLE
Response hook

### PR DESCRIPTION
**Pull request by ineiti**
Also changed `service.Cosi` to `service.CoSi`.

### CODE REVIEW

**Comment by liamsi** [comment]
Merge conflicts -> back to WIP

File: protocol/cosi.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ type CommitmentHook func(in []abstract.Point) error
 type ChallengeHook func(ch abstract.Scalar) error
 
 // ResponseHook allows for handling what should happen when all
-// responses are received
+// responses are received and our response is calculated
```
Comment (liamsi) [inline protocol/cosi.go:79]: nitpick: maybe "called" instead of "calculated"

Comment (liamsi) [inline protocol/cosi.go:79]: No, It's OK. I though it was about the callback, I mean handler ;-) and it's about the actual repsonse (which is indeed calculated)

**Comment by liamsi** [comment]
OK, to merge.

**Comment by ineiti** [comment]
Honor to you ;)

**Pull request merged**